### PR TITLE
Heatmap cleanup

### PR DESCRIPTION
**Pull request by mortonjt**
The two main things that are done in this PR

1. Making line width an exposed parameter
2. Making matplotlib colormaps an exposed parameter

Here's some examples of how to run the qiime commands and their resulting outputs in view.qiime2.org
```
qiime gneiss dendrogram-heatmap \
    --i-table 88soils_composition.biom.qza \
    --i-tree ph_tree.nwk.qza \
    --m-metadata-file 88soils_rounded_metadata.txt \
    --m-metadata-category "ph_rounded" \
    --o-visualization "viridis_heatmap"
```
screen shot 2017-04-09 at 6 03 47 pm

```
qiime gneiss dendrogram-heatmap \
    --i-table 88soils_composition.biom.qza \
    --i-tree ph_tree.nwk.qza \
    --m-metadata-file 88soils_rounded_metadata.txt \
    --m-metadata-category ph_rounded \
    --p-color-map "seismic" \
    --o-visualization "seismic_heatmap"
```
screen shot 2017-04-09 at 6 03 51 pm

```
qiime gneiss dendrogram-heatmap \
    --i-table 88soils_composition.biom.qza \
    --i-tree ph_tree.nwk.qza \
    --m-metadata-file 88soils_rounded_metadata.txt \
    --m-metadata-category ph_rounded \
    --p-color-map "seismic" \
    --p-method "log" \
    --o-visualization "log_seismic_heatmap"
```
screen shot 2017-04-09 at 6 03 58 pm

And here's the help menu
```
tests-MacBook-Pro-4:gneiss mortonjt$ qiime gneiss dendrogram-heatmap --help
Usage: qiime gneiss dendrogram-heatmap [OPTIONS]

  Visualize the feature tables as a heatmap. with samples sorted along a
  specified metadata category and features clustered together specified by
  the tree.In addition, the heatmap values are clr transformed.

Options:
  --i-table PATH                  Artifact: FeatureTable[Composition]
                                  [required]
                                  The feature table that will be
                                  plotted as a heatmap. This table is assumed
                                  to have strictly positive values.
  --i-tree PATH                   Artifact: Phylogeny[Rooted]  [required]
                                  A
                                  hierarchy of feature identifiers where each
                                  tipcorresponds to the feature identifiers in
                                  the table. This tree can contain tip ids
                                  that are not present in the table, but all
                                  feature ids in the table must be present in
                                  this tree.
  --m-metadata-file PATH          Metadata mapping file  [required]
  --m-metadata-category TEXT      Category from metadata mapping file
                                  [required]
  --p-ndim INTEGER                [default: 10]
                                  Number of dimensions to
                                  highlight.
  --p-method [log|clr]            [default: clr]
                                  Specifies how the data should
                                  be normalized for display.Options include
                                  'log' or 'clr' (default='clr').
  --p-color-map [PuBu|ocean|PuRd|Vega20|Set1|Greens|Spectral|RdPu|magma|inferno|gist_stern|gist_ncar|BuPu|gist_rainbow|spring|Reds|cool|hot|Paired|Pastel2|Set2|RdYlBu|OrRd|BuGn|Vega20c|brg|PuBuGn
                                  [default: viridis]
                                  Specifies the color map
                                  for plotting the heatmap. See https://matplo
                                  tlib.org/examples/color/colormaps_reference.
                                  html for more details.
  --o-visualization PATH          Artifact: Visualization  [required if not
                                  passing --output-dir]
  --output-dir DIRECTORY          Output unspecified results to a directory
  --cmd-config PATH               Use config file for command options
  --verbose                       Display verbose output to stdout and/or
                                  stderr during execution of this action.
                                  [default: False]
  --help                          Show this message and exit.
```
You'll see that we are dumping all of the matplotlib colormaps as options.  So the user can't view all of the possible options unless the following is given

```
qiime gneiss dendrogram-heatmap 2> help.txt
```

Note that I'm not sure how to write up tests for the `method` or `color-map` parameters.

### CODE REVIEW

File: gneiss/plot/_heatmap.py [hunk 1]
```diff
@@ -31,6 +31,10 @@ def heatmap(table, tree, mdvar, highlights=None,
         subtree and the other for the right subtree highlight.
         The first color will always correspond to the left subtree,
         and the second color will always correspond to the right subtree.
+    cmap : str
+        Species the matplotlib colormap for the heatmap (default='viridis')
```
Comment (tanaes) [inline gneiss/plot/_heatmap.py]: Species -> Specifies

File: gneiss/plot/_plot.py [hunk 2]
```diff
@@ -387,10 +413,15 @@ def dendrogram_heatmap(output_dir: str, table: pd.DataFrame,
                  'present in this tree.')},
     parameter_descriptions={
         'metadata': ('Metadata to group the samples. '),
-        'ndim': 'Number of dimensions to highlight.'},
+        'ndim': 'Number of dimensions to highlight.',
```
Comment (tanaes) [inline gneiss/plot/_plot.py]: So this is always just highlighting the `n` most basal splits in the tree, right? 

In the future do you want to expose selective highlighting of e.g. most significant splits, or prefer to keep those int he boxplot visualizations?

Comment (mortonjt) [inline gneiss/plot/_plot.py]: Exactly! Just using the `n` most basal splits in the command line interface.  For the python interface, you can specify what ever splits you are interested in (see [here](https://github.com/biocore/gneiss/blob/master/gneiss/plot/_heatmap.py#L28)).

The reason why we are doing this in the CLI, is because it isn't clear how to best input the splits you are interested in, along with the colors.  

tl;dr this is available in the python interface, but not the q2 interface.

**Comment by mortonjt** [comment]
@tanaes do you think there is anything else needed for this PR?  If not, could you merge it?  Thanks!

**Comment by mortonjt** [comment]
Same with this PR.  I'm going to merge this since this has already been approved.
